examples/images_contours_and_fields/specgram_demo.py

```diff
@@ -32,9 +31,7 @@
 # the frequency vector, bins are the centers of the time bins in which
 # the power is computed, and im is the matplotlib.image.AxesImage
 # instance
-
-ax1 = plt.subplot(211)
-plt.plot(t, x)
-plt.subplot(212, sharex=ax1)
-Pxx, freqs, bins, im = plt.specgram(x, NFFT=NFFT, Fs=Fs, noverlap=900)
+fig, (ax1, ax2) = plt.subplots(nrows=2)
+line, = ax1.plot(t, x)
```

Comment:
`line` is not used afterwards, so it should be left out. Same with the return values on the following line.

Reply:
I'm going to push back on this a bit. My gut feeling is that users -- especially new users -- are more likely to read these example than the docstrings.

So I think we should use this examples to be as explicit as possible about what each method is doing. 

There are a lot of questions on SO along the lines of "how do I get an artist that's on my plot". While, there's a decent way to do that after the fact, I think it'd be nice to get users in the habbit of capturing things sooner rather than searching for them later.

Reply:
I'm 100% with you that users are learning from the examples. Therefore a big :+1: for updating the examples.

My conclusion from learning by example is that the examples should be similar to real-world uses. I fear that if all the examples had  `line = ax.plot(x, y)`, you'll find a lot of people having exactly that in their scripts, even if they don't use `line` at all - at least that's my experience with less-experienced users.

I'm fine, if you want to keep the return values on `specgram()`. That's what the example is about. If so, it would be good to move the related comment from a few lines above directly to the function call. Something like:

~~~
Pxx, freqs, bins, im = ax2.specgram(x, NFFT=NFFT, Fs=Fs, noverlap=900)
# returned values:
# - Pxx: the periodograms
# - freqs: the frequency vector
# - bins: the centers of the time bins
# - im: the matplotlib.image.AxesImage instance representing the data in the plot
~~~

However, since this is not about `plot()`, I wouldn't add the unused `line`.

Reply:
Well said. I'm on aboard.